File: backend/src/main/java/protasks/backend/Message/Message.java
package protasks.backend.Message;

import com.fasterxml.jackson.annotation.JsonView;
import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder;
import protasks.backend.Board.BoardUsersPermRel;
import protasks.backend.Task.Task;
import protasks.backend.user.User;

import javax.persistence.*;
import java.util.ArrayList;
import java.util.Date;
import java.util.List;

@Entity
public class Message {
    public interface MessageBasicInfo{}
    public interface MessageExtendedInfo{}

    @JsonView(MessageBasicInfo.class)
    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    @Column(name="Id")
    private long id;

    @JsonView(MessageBasicInfo.class)
    @Column(name="Body")
    private String body;

    @JsonView(MessageBasicInfo.class)
    @ManyToOne
    private User user;

    @JsonView(MessageExtendedInfo.class)
    @ManyToOne
    private Task task;

    @JsonView(MessageBasicInfo.class)
    @Column(name="write_date")
    @Temporal(TemporalType.TIMESTAMP)
    private Date write_date;

    @JsonView(MessageBasicInfo.class)
    @Column(name="create_date")
    @Temporal(TemporalType.TIMESTAMP)
    private Date create_date;


    public Message() {
    }

    public Message(String body, User user, Task task) {
        this.body = body;
        this.user = user;
        this.task = task;
    }

    public long getId() {
        return id;
    }

    public void setId(long id) {
        this.id = id;
    }

    public String getBody() {
        return body;
    }

    public void setBody(String body) {
        this.body = body;
    }

    public User getUser() {
        return user;
    }

    public void setUser(User user) {
        this.user = user;
    }

    public Task getTask() {
        return task;
    }

    public void setTask(Task task) {
        this.task = task;
    }

    public void setWrite_date(Date write_date) {
        this.write_date = write_date;
    }

    public void setCreate_date(Date create_date) {
        this.create_date = create_date;
    }

}

File: app/src/main/java/com/example/protasks/models/Message.kt
package com.example.protasks.models

import com.google.gson.annotations.Expose
import com.google.gson.annotations.SerializedName

class Message {
    @Expose
    @SerializedName("id")
    private var id: Long = 0

    @Expose
    @SerializedName("task")
    private var task: Task? = null

    @Expose
    @SerializedName("body")
    private var body: String? = null

    @Expose
    @SerializedName("user")
    private var user: User? = null

    fun setId(id: Long) {
        this.id = id
    }


    fun getId(): Long {
        return id
    }


    fun getBody(): String? {
        return body
    }

    fun setBody(body: String?) {
        this.body = body
    }

    fun getTask(): Task? {
        return task
    }

    fun setTask(task: Task) {
        this.task = task
    }

    fun getUser(): User? {
        return user
    }

    fun setUser(user: User) {
        this.user = user
    }


}